frontend/src/components/NavBar.js

```diff
@@ -37,7 +44,12 @@ const NavBar = () => {
   }
 
   const _renderAccountIcon = () => {
-    if (!isUserLoggedIn()) {
+    if (
+      !isUserLoggedIn() &&
+      (location.pathname === baseUrl ||
+        location.pathname === signupUrl ||
+        location.pathname === loginUrl)
```

Comment:
This works fine! But just a small code quality suggestion to show extensibility:
- We can define a list of URLs that won't show the account icon somewhere (e.g. routeConstants.js perhaps)
```
const publicRoutes = [baseUrl, signupUrl, loginUrl]
```

- We can refactor the check here:
```
if (... && publicRoutes.includes(location.pathname) { ... }
```

This way, we can add 5, 10, or 20 more routes and it won't become 20 lines of `location.pathname === <URL> || ....`

Reply:
Ahh, this is a good suggestion. I will change it!